RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,16 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    throw cms::Exception("PhotonProducer") 
+      << "Error! Can't get the preshowerEcalHits";
+  }
```

Comment:
Hi @Sam-Harper ,

I was following along with what was done in the GEDPhotonProducer: https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc#L293-L301

I can make the change either way.  let me know what you prefer.